### ISSUE TITLE
`--side server` not including dependencies if tagged as client-side

### ISSUE THREAD

**Issue by abigailwillow** [packwiz/packwiz#291]
For example [Chipped](https://modrinth.com/mod/chipped) depends on [Athena](https://modrinth.com/mod/athena-ctm). Chipped is properly labeled as **both**, but Athena is mislabeled as **client** mod.

During the building of a modpack it should probably include any mods that are dependencies of other mods which are included in that build, regardless of which side they're labeled as.

**Comment by comp500** [comment]
Is Athena mislabeled? My assumption would be that a connected textures mod only needs to be installed on the client side, even though the blocks are added by a mod that is required on both sides. I do think there are other cases where this would make sense too, like config GUI mods.

**Comment by abigailwillow** [comment]
That's a good point. Chipped should probably not depend on Athena for its server-side mod. I'm unsure how this works exactly since I'm unfamiliar with mod development. Perhaps it is a technical limitation with dependencies? Perhaps to ensure compatibility, it could be considered to include dependencies, even if they are not strictly necessary for the server-side.

**Comment by comp500** [comment]
Yeah I don't think it's possible to specify "requires this mod but only on the client side".
The most common mod compatibility problem with sidedness is that a client-side only mod has been run on the server and expects client-side code to exist - so respecting what a client-side mod is marked as on Modrinth is preferable.

In an ideal world, all mods would be able to exist on both sides and we wouldn't have to treat client/server side mods differently - it's pretty easy for mod authors to just mark their mod as client-only in NeoForge/Fabric/Quilt and it just won't be loaded on the server.

Either way, I think the behaviour packwiz has at the moment is good enough, unless you can think of a way to improve it?

**Comment by abigailwillow** [comment]
The point is when building a pack with a clientside dependency packwiz using `--side server` right now the server won't start because of the missing dependency, completely invalidating the point of the parameter.

My suggested solution is to always include the dependency mod, regardless of whether it is client or server side.

**Comment by comp500** [comment]
Ahh I see what you mean now - the dependency is also set in the mod metadata itself, so it won't launch without the dependency?

That makes sense... I think in future the mod metadata can be something packwiz scans, to determine if dependencies are needed? For now it does make sense to just add dependencies of side=both mods as side=both.

**Comment by TheEpicBlock** [comment]
With #284 fixed, Modrinth metadata is now properly interpreted. This means this situation should no longer really happen unless someone sets the completely wrong Modrinth environment. In that situations it's really up to the projects, any workaround in packwiz is moot since there are legit situations where a "both" mod might depend on a client/server one
